src/check_lines.rs

```diff
@@ -190,6 +190,25 @@ pub fn check_lines(
         }
     }
 
+    // Enfore newlines at end of files. Don't check if noqa enabled.
+    if settings.enabled.contains(&CheckCode::W292) && !enforce_noqa {
+        // If the file contains a newline at the end of it the last
+        // line would contain an empty string slice.
+        let line = lines.last().unwrap_or(&"");
+
+        let check = Check::new(
+            CheckKind::NoNewLineAtEndOfFile,
+            Range {
+                location: Location::new(lines.len(), line.len() + 1),
+                end_location: Location::new(lines.len(), line.len() + 1),
+            },
+        );
```

Comment:
Mind moving this into `if !line.is_empty() { ... }` below? No need to instantiate it unless we're going to push it.